tests/arduino_lis3dh/Makefile

```diff
@@ -0,0 +1,12 @@
+include ../Makefile.tests_common
+
+BOARD_INSUFFICIENT_MEMORY := arduino-duemilanove arduino-leonardo arduino-nano \
+                             arduino-uno nucleo-f031k6
+
+USEPKG += arduino_lis3dh
+USEMODULE += arduino_lib
+USEMODULE += cpp11-compat
+USEMODULE += periph_i2c
+USEMODULE += periph_spi
```

Comment:
Maybe conditionally add the right module to the build:

ARDUINO_LIS3DH_MODE ?= i2c

ifeq (i2c,$(ARDUINO_LIS3DH_MODE))
  FEATURES_REQUIRED += periph_i2c
else ifeq (spi,$(ARDUINO_LIS3DH_MODE))
  FEATURES_REQUIRED += periph_spi
  CFLAGS += -DARDUINO_LIS3DH_USE_SPI
endif

Reply:
After trying a lot, it is not possible to define such dependencies without changing the original source completely by patching what is not the in intention of package. The reason is that both interfaces are implemented by the driver which are not selected statically but by member values.